SimG4CMS/Tracker/interface/TrackerG4SimHitNumberingScheme.h

```diff
@@ -1,27 +1,24 @@
 #ifndef SimG4CMS_TrackerG4SimHitNumberingScheme_H
 #define SimG4CMS_TrackerG4SimHitNumberingScheme_H
 
-#include <vector>
+#include "SimG4CMS/Tracker/interface/TouchableToHistory.h"
+#include "G4VTouchable.hh"
 
-class TouchableToHistory;
-class G4VPhysicalVolume;
-class G4VTouchable;
 class DDCompactView;
 class GeometricDet;
 
 class TrackerG4SimHitNumberingScheme 
 {
 public:
-  typedef std::vector<int> nav_type;
+
   TrackerG4SimHitNumberingScheme(const DDCompactView&, const GeometricDet&);
-  ~TrackerG4SimHitNumberingScheme(){clear();}
+  ~TrackerG4SimHitNumberingScheme();
     
-  void clear();
+  inline unsigned int g4ToNumberingScheme(const G4VTouchable* touch)
+  { return ts->touchableToInt(touch); }
 
-  unsigned int g4ToNumberingScheme(const G4VTouchable*);
-  
 private:
-    TouchableToHistory * ts;
+  TouchableToHistory * ts;
```

Comment:
what about unique_ptr or even just by holding it value?
the copy behavior of this class shall be better constrained...
btw: what is the real difference between a TrackerG4SimHitNumberingScheme  and a TouchableToHistory ??